packages/analytics-js/src/components/pluginsManager/PluginsManager.ts

```diff
@@ -93,10 +96,19 @@ class PluginsManager implements IPluginsManager {
       return [];
     }
 
+    // Filter deprecated plugins
+    pluginsToLoadFromConfig = pluginsToLoadFromConfig.filter(pluginName => {
+      if (deprecatedPluginsList.includes(pluginName)) {
+        this.logger?.warn(DEPRECATED_PLUGIN_WARNING(PLUGINS_MANAGER, pluginName));
+        return true;
+      }
+      return false;
```

Comment:
The logic should be the opposite. You need to "exclude" (filter) the deprecated plugins.

It seems like no testing was done!

At the very least, you could have added unit tests.
You can extract this logic into a function if that makes it easier to test.

Reply:
Why no tests were added?